src/plan_expand_hypertable.c

```diff
@@ -70,6 +72,124 @@ is_chunk_exclusion_func(Expr *node)
 	return false;
 }
 
+static bool
+is_time_bucket_function(Expr *node)
+{
+	if (IsA(node, FuncExpr) &&
+		strncmp(get_func_name(castNode(FuncExpr, node)->funcid), "time_bucket", NAMEDATALEN) == 0)
+		return true;
+
+	return false;
+}
+
+/*
+ * Transform time_bucket calls of the following form in WHERE clause:
+ *
+ * time_bucket(width, column) > value
+ * column > value - width
+ *
+ * time_bucket(width, column) < value
+ * column < value + width
+ *
+ * This type of WHERE clause is used when querying continuous aggregation
+ * views.
```

Comment:
As an outside reviewer, I found Josh's derivation above much easier to verify. Please include it (or something like it) in the comment.

Reply:
Especially the >= vs > is easier to follow